thevoid/connection.cpp

```diff
@@ -23,6 +23,66 @@
 #include "stream_p.hpp"
 #include "stockreplies_p.hpp"
 
+namespace {
+
+const long int USECS_IN_SEC = 1000 * 1000;
+const long int NSECS_IN_SEC = USECS_IN_SEC * 1000;
+
+struct timespec& operator+= (struct timespec& lhs, const struct timespec& rhs) {
+	lhs.tv_sec += rhs.tv_sec;
+	lhs.tv_nsec += rhs.tv_nsec;
+
+	if (lhs.tv_nsec >= NSECS_IN_SEC) {
+		lhs.tv_sec += 1;
+		lhs.tv_nsec -= NSECS_IN_SEC;
+	}
+
+	return lhs;
+}
+
+struct timespec operator+ (struct timespec lhs, const struct timespec& rhs) {
+	return lhs += rhs;
+}
+
+struct timespec& operator-= (struct timespec& lhs, const struct timespec& rhs) {
+	lhs.tv_sec -= rhs.tv_sec;
+	lhs.tv_nsec -= rhs.tv_nsec;
+
+	if (lhs.tv_nsec < 0) {
+		lhs.tv_sec -= 1;
+		lhs.tv_nsec += NSECS_IN_SEC;
+	}
+
+	if (lhs.tv_sec < 0) {
+		lhs = {0, 0};
+	}
+
+	return lhs;
+}
+
+struct timespec operator- (struct timespec lhs, const struct timespec& rhs) {
+	return lhs -= rhs;
+}
+
+long int timespec_to_usec(const struct timespec& t) {
+	return t.tv_sec * USECS_IN_SEC + t.tv_nsec / 1000;
+}
+
+struct timespec gettime_now() {
+	struct timespec now;
+
+#ifdef CLOCK_MONOTONIC_RAW
+	if (clock_gettime(CLOCK_MONOTONIC_RAW, &now) == 0) {
+		return now;
+	}
+#endif
+
+	clock_gettime(CLOCK_MONOTONIC, &now);
+	return now;
+}
```

Comment:
I read a bit about the difference between CLOCK_MONOTONIC_RAW (CMR) and CLOCK_MONOTONIC (CM). And noticed it is not undoubted for me that CMR preferable than CM in this case.

This refs I found quite useful:
https://blog.habets.se/2010/09/gettimeofday-should-never-be-used-to-measure-time
http://www.spinics.net/lists/linux-man/msg00973.html
http://www.reddit.com/comments/y3mxv

As I understood CMR is better to use for frequent events that occur more frequently than NTP adjusts time, and CM is better to use for long term measurements.

Can you make it more clear?